src/shared/common/ScaleBar.js

```diff
@@ -6,12 +6,19 @@ import { Path, Shape, Surface, Text } from '@react-native-community/art';
 import GLOBAL from '../Globals';
 
 type Props = {
+    // whether to shift the scale bar a little bit up from the bottom
+    alignToBottom: boolean,
     latitude: number,
+    // true if we should use the screen_width to size the scale bar instead of
+    // using the tile size. This is useful for building footprint projects
+    // with TMS imagery as we use a few tricks there to cover the entire screen
+    // with imagery
+    useScreenWidth: boolean,
     visible: boolean,
     zoomLevel: number,
 };
 
-const getScaleBar = (meters, feet, tileWidth) => {
+const getScaleBar = (meters, feet, tileWidth, referenceSize) => {
```

Comment:
just to understand better: for TMS projects there will be referenceSize = tileWidth, but for Building Footprint this will be two different values

Reply:
Yup. I think the comment about `useScreenWidth` at the top and the code below should make it clear. If not, just comment back here, and I'll try to elaborate.